src/rgw/rgw_op.cc

```diff
@@ -4550,18 +4548,19 @@ void RGWBulkDelete::pre_exec()
 
 void RGWBulkDelete::execute()
 {
-  deleter = std::unique_ptr<Deleter>(new Deleter(store, s));
+  deleter = std::unique_ptr<Deleter>(new RGWBulkDelete::Deleter(store, s));
```

Comment:
another option is to use move semantics, i.e. the `RGWBulkDelete` constructor would default-construct a `Deleter`, and `execute()` would use move-assignment to replace it with a newly constructed `Deleter` once it has the `req_state`